internal/component/prometheus/scrape/scrape.go

```diff
@@ -126,11 +128,15 @@ type Component struct {
 
 	reloadTargets chan struct{}
 
-	mut          sync.RWMutex
-	args         Arguments
-	scraper      *scrape.Manager
-	appendable   *prometheus.Fanout
-	targetsGauge client_prometheus.Gauge
+	mut                 sync.RWMutex
+	args                Arguments
+	scraper             *scrape.Manager
+	appendable          *prometheus.Fanout
+	targetsGauge        client_prometheus.Gauge
+	movedTargetsCounter client_prometheus.Counter
```

Comment:
nit: do targetsGauge and movedTargetsCounter need to be protected by mut? AFAIK client_prometheus metrics are Goroutine safe. 

I recommend a pattern where a stanza of struct fields starts with a mut to indicate that everything within that stanza needs protection from that mut; if that isn't true here then I'd suggest considering moving the metric fields to a new stanza.